libraries/AP_Common/AP_ExpandingArray.h

```diff
@@ -0,0 +1,115 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <AP_Common/AP_Common.h>
+
+template <typename T>
+class AP_ExpandingArray
+{
+public:
+
+    AP_ExpandingArray<T>() :
+        chunk_size(50)
+    {   // create one chunk
+        expand(1);
+    }
+
+    AP_ExpandingArray<T>(uint16_t num_chunks, uint16_t elements_per_chunk) :
```

Comment:
I guess this is two separate suggestions.

- requiring the user to always specify the number of elements-per-chunk sounds fine to me.  I'll remove the default constructor.
- the idea for allowing the user to specify the number of chunks was to get the memory allocation out of the way earlier.  I.e. at startup before arming rather than at run-time.  That could be handled by the caller though pretty easily so I'll remove this.